tools/modules/update.sh

```diff
@@ -16,28 +16,30 @@
 # update_modules.sh updates the repositories for external code (to the precise
 # revisions used for testing).
 #
-# If the single argument `--git_only` is passed, the repositories will be
+# If the single argument `--fetch_only` is passed, the repositories will be
 # updated to the pinned versions without configuring or building them.
 # If the argument `--build_only` is passed, the script will assume that the
 # repositories are at the correct version and will configure and build them.
 
-git_maybe_clone() {
-  local repo="$1"
+wget_copy_archive() {
+  local target="$1"
   local dir="$2"
-  if [[ ! -d "$dir/.git" ]]; then
-    git clone "$repo" "$dir"
-  fi
-}
+  local sha="$3"
 
-git_checkout_sha() {
-  local repo="$1"
-  local sha="$2"
-  cd "$repo"
-  if [[ "$sha" != "$(git rev-parse HEAD)" ]]; then
-    git fetch origin master
-    git checkout -f "$sha"
+  if [[ ! -f "$dir/$sha.sentinel" ]]; then
```

Comment:
Another approach that might be safer would be to always unpack into a directory named for the SHA, and just update a symlink pointing to that directory, e.g., `llvm -> a0c310991b…30c0`

Reply:
(though that would mean these directories would accumulate over time, so maybe that's not great either—though clobbering a whole directory structure from a script always gives me pause)

Reply:
Yup, my two exact thoughts.  Shahms also independently had the same idea.

I agree that clobbering the directory structure from script doesn't sit well with me, but gradually accumulating 500MBs of stuff every dep doesn't sit well with me either.

I'm not convinced which way is better yet, this is just the one I settled on so far.